{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Samina's Gene pairs Dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('samina_results.csv')[1:].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID_gene1</th>\n",
       "      <th>ID_gene2</th>\n",
       "      <th>ch_gene1</th>\n",
       "      <th>ch_gene2</th>\n",
       "      <th>strand_gene1</th>\n",
       "      <th>strand_gene2</th>\n",
       "      <th>type_gene1</th>\n",
       "      <th>type_gene2</th>\n",
       "      <th>name_gene1</th>\n",
       "      <th>name_gene2</th>\n",
       "      <th>start_end_gene1</th>\n",
       "      <th>start_end_gene2</th>\n",
       "      <th>genename1</th>\n",
       "      <th>genename2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ST0002</td>\n",
       "      <td>ST3638</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>CUTs</td>\n",
       "      <td>CUTs</td>\n",
       "      <td>CUT001</td>\n",
       "      <td>CUT437</td>\n",
       "      <td>(30073,30905)</td>\n",
       "      <td>(30533,30893)</td>\n",
       "      <td>unknown</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ST0003</td>\n",
       "      <td>ST3639</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>SUTs</td>\n",
       "      <td>YAL062W</td>\n",
       "      <td>SUT434</td>\n",
       "      <td>(31153,32985)</td>\n",
       "      <td>(31485,32749)</td>\n",
       "      <td>Glutamate DeHydrogenase</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ST3640</td>\n",
       "      <td>ST0004</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "      <td>+</td>\n",
       "      <td>SUTs</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>SUT435</td>\n",
       "      <td>YAL061W</td>\n",
       "      <td>(33077,34381)</td>\n",
       "      <td>(33361,34897)</td>\n",
       "      <td>unknown</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ST0004</td>\n",
       "      <td>ST3641</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>CUTs</td>\n",
       "      <td>YAL061W</td>\n",
       "      <td>CUT438</td>\n",
       "      <td>(33361,34897)</td>\n",
       "      <td>(34381,34749)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ST0005</td>\n",
       "      <td>ST3642</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>+</td>\n",
       "      <td>-</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>CUTs</td>\n",
       "      <td>YAL060W</td>\n",
       "      <td>CUT439</td>\n",
       "      <td>(35097,36393)</td>\n",
       "      <td>(35797,36349)</td>\n",
       "      <td>Butanediol DeHydrogenase</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  ID_gene1 ID_gene2  ch_gene1  ch_gene2 strand_gene1 strand_gene2 type_gene1  \\\n",
       "1   ST0002   ST3638       1.0       1.0            +            -       CUTs   \n",
       "2   ST0003   ST3639       1.0       1.0            +            -      ORF-T   \n",
       "3   ST3640   ST0004       1.0       1.0            -            +       SUTs   \n",
       "4   ST0004   ST3641       1.0       1.0            +            -      ORF-T   \n",
       "5   ST0005   ST3642       1.0       1.0            +            -      ORF-T   \n",
       "\n",
       "  type_gene2 name_gene1 name_gene2 start_end_gene1 start_end_gene2  \\\n",
       "1       CUTs     CUT001     CUT437   (30073,30905)   (30533,30893)   \n",
       "2       SUTs    YAL062W     SUT434   (31153,32985)   (31485,32749)   \n",
       "3      ORF-T     SUT435    YAL061W   (33077,34381)   (33361,34897)   \n",
       "4       CUTs    YAL061W     CUT438   (33361,34897)   (34381,34749)   \n",
       "5       CUTs    YAL060W     CUT439   (35097,36393)   (35797,36349)   \n",
       "\n",
       "                  genename1 genename2  \n",
       "1                   unknown   unknown  \n",
       "2   Glutamate DeHydrogenase   unknown  \n",
       "3                   unknown       NaN  \n",
       "4                       NaN   unknown  \n",
       "5  Butanediol DeHydrogenase   unknown  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['ID_gene1', 'ID_gene2', 'ch_gene1', 'ch_gene2', 'strand_gene1',\n",
       "       'strand_gene2', 'type_gene1', 'type_gene2', 'name_gene1', 'name_gene2',\n",
       "       'start_end_gene1', 'start_end_gene2', 'genename1', 'genename2'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 2954 entries, 1 to 2954\n",
      "Data columns (total 14 columns):\n",
      " #   Column           Non-Null Count  Dtype  \n",
      "---  ------           --------------  -----  \n",
      " 0   ID_gene1         2954 non-null   object \n",
      " 1   ID_gene2         2954 non-null   object \n",
      " 2   ch_gene1         2954 non-null   float64\n",
      " 3   ch_gene2         2954 non-null   float64\n",
      " 4   strand_gene1     2954 non-null   object \n",
      " 5   strand_gene2     2954 non-null   object \n",
      " 6   type_gene1       2954 non-null   object \n",
      " 7   type_gene2       2954 non-null   object \n",
      " 8   name_gene1       2954 non-null   object \n",
      " 9   name_gene2       2954 non-null   object \n",
      " 10  start_end_gene1  2954 non-null   object \n",
      " 11  start_end_gene2  2954 non-null   object \n",
      " 12  genename1        2706 non-null   object \n",
      " 13  genename2        2743 non-null   object \n",
      "dtypes: float64(2), object(12)\n",
      "memory usage: 323.2+ KB\n"
     ]
    }
   ],
   "source": [
    "df.info()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Transforming paris of genes into single genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "first_gene_columns = ['ID_gene1','ch_gene1','strand_gene1','type_gene1','name_gene1','start_end_gene1','genename1']\n",
    "second_gene_columns = ['ID_gene2','ch_gene2','strand_gene2','type_gene2','name_gene2','start_end_gene2','genename2']\n",
    "correct_column_names = ['id','chrom','strand','type','name','start_end','genname']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "first_genes = df[first_gene_columns]\n",
    "second_genes = df[second_gene_columns]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "corrected_first_genes_df = first_genes.rename(columns={first_gene_columns[i]:correct_column_names[i] for i in range(7)})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "corrected_second_genes_df = second_genes.rename(columns={second_gene_columns[i]:correct_column_names[i] for i in range(7)})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_df = pd.concat([corrected_first_genes_df,corrected_second_genes_df]).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Removing Duplicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_df = genes_df.drop_duplicates().reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_df['chrom'] = genes_df['chrom'].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>strand</th>\n",
       "      <th>type</th>\n",
       "      <th>name</th>\n",
       "      <th>start_end</th>\n",
       "      <th>genname</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ST0002</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>CUTs</td>\n",
       "      <td>CUT001</td>\n",
       "      <td>(30073,30905)</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ST0003</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>YAL062W</td>\n",
       "      <td>(31153,32985)</td>\n",
       "      <td>Glutamate DeHydrogenase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ST3640</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>SUTs</td>\n",
       "      <td>SUT435</td>\n",
       "      <td>(33077,34381)</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ST0004</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>YAL061W</td>\n",
       "      <td>(33361,34897)</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ST0005</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>YAL060W</td>\n",
       "      <td>(35097,36393)</td>\n",
       "      <td>Butanediol DeHydrogenase</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4940</th>\n",
       "      <td>ST7262</td>\n",
       "      <td>16</td>\n",
       "      <td>-</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>YPR188C</td>\n",
       "      <td>(911823,912511)</td>\n",
       "      <td>Myo1p Light Chain</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4941</th>\n",
       "      <td>ST3625</td>\n",
       "      <td>16</td>\n",
       "      <td>+</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>YPR192W</td>\n",
       "      <td>(921755,922827)</td>\n",
       "      <td>AQuaporin from Yeast</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4942</th>\n",
       "      <td>ST3627</td>\n",
       "      <td>16</td>\n",
       "      <td>+</td>\n",
       "      <td>SUTs</td>\n",
       "      <td>SUT429</td>\n",
       "      <td>(927659,928459)</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4943</th>\n",
       "      <td>ST3629</td>\n",
       "      <td>16</td>\n",
       "      <td>+</td>\n",
       "      <td>other</td>\n",
       "      <td>YPR196W, YPRWtau4</td>\n",
       "      <td>(931203,933179)</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4944</th>\n",
       "      <td>ST3633</td>\n",
       "      <td>16</td>\n",
       "      <td>+</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>YPR201W</td>\n",
       "      <td>(939683,942347)</td>\n",
       "      <td>ARsenicals Resistance</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4945 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          id  chrom strand   type               name        start_end  \\\n",
       "0     ST0002      1      +   CUTs             CUT001    (30073,30905)   \n",
       "1     ST0003      1      +  ORF-T            YAL062W    (31153,32985)   \n",
       "2     ST3640      1      -   SUTs             SUT435    (33077,34381)   \n",
       "3     ST0004      1      +  ORF-T            YAL061W    (33361,34897)   \n",
       "4     ST0005      1      +  ORF-T            YAL060W    (35097,36393)   \n",
       "...      ...    ...    ...    ...                ...              ...   \n",
       "4940  ST7262     16      -  ORF-T            YPR188C  (911823,912511)   \n",
       "4941  ST3625     16      +  ORF-T            YPR192W  (921755,922827)   \n",
       "4942  ST3627     16      +   SUTs             SUT429  (927659,928459)   \n",
       "4943  ST3629     16      +  other  YPR196W, YPRWtau4  (931203,933179)   \n",
       "4944  ST3633     16      +  ORF-T            YPR201W  (939683,942347)   \n",
       "\n",
       "                       genname  \n",
       "0                      unknown  \n",
       "1      Glutamate DeHydrogenase  \n",
       "2                      unknown  \n",
       "3                          NaN  \n",
       "4     Butanediol DeHydrogenase  \n",
       "...                        ...  \n",
       "4940         Myo1p Light Chain  \n",
       "4941      AQuaporin from Yeast  \n",
       "4942                   unknown  \n",
       "4943                   unknown  \n",
       "4944     ARsenicals Resistance  \n",
       "\n",
       "[4945 rows x 7 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Splitting start and end"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_df['start'] = genes_df['start_end'].apply(lambda x: int(x[1:-1].split(',')[0]))\n",
    "genes_df['end'] = genes_df['start_end'].apply(lambda x: int(x[1:-1].split(',')[1]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_df.drop(columns=['start_end'],inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>strand</th>\n",
       "      <th>type</th>\n",
       "      <th>name</th>\n",
       "      <th>genname</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ST0002</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>CUTs</td>\n",
       "      <td>CUT001</td>\n",
       "      <td>unknown</td>\n",
       "      <td>30073</td>\n",
       "      <td>30905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ST0003</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>YAL062W</td>\n",
       "      <td>Glutamate DeHydrogenase</td>\n",
       "      <td>31153</td>\n",
       "      <td>32985</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ST3640</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>SUTs</td>\n",
       "      <td>SUT435</td>\n",
       "      <td>unknown</td>\n",
       "      <td>33077</td>\n",
       "      <td>34381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ST0004</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>YAL061W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33361</td>\n",
       "      <td>34897</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ST0005</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>YAL060W</td>\n",
       "      <td>Butanediol DeHydrogenase</td>\n",
       "      <td>35097</td>\n",
       "      <td>36393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4940</th>\n",
       "      <td>ST7262</td>\n",
       "      <td>16</td>\n",
       "      <td>-</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>YPR188C</td>\n",
       "      <td>Myo1p Light Chain</td>\n",
       "      <td>911823</td>\n",
       "      <td>912511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4941</th>\n",
       "      <td>ST3625</td>\n",
       "      <td>16</td>\n",
       "      <td>+</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>YPR192W</td>\n",
       "      <td>AQuaporin from Yeast</td>\n",
       "      <td>921755</td>\n",
       "      <td>922827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4942</th>\n",
       "      <td>ST3627</td>\n",
       "      <td>16</td>\n",
       "      <td>+</td>\n",
       "      <td>SUTs</td>\n",
       "      <td>SUT429</td>\n",
       "      <td>unknown</td>\n",
       "      <td>927659</td>\n",
       "      <td>928459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4943</th>\n",
       "      <td>ST3629</td>\n",
       "      <td>16</td>\n",
       "      <td>+</td>\n",
       "      <td>other</td>\n",
       "      <td>YPR196W, YPRWtau4</td>\n",
       "      <td>unknown</td>\n",
       "      <td>931203</td>\n",
       "      <td>933179</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4944</th>\n",
       "      <td>ST3633</td>\n",
       "      <td>16</td>\n",
       "      <td>+</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>YPR201W</td>\n",
       "      <td>ARsenicals Resistance</td>\n",
       "      <td>939683</td>\n",
       "      <td>942347</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4945 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          id  chrom strand   type               name  \\\n",
       "0     ST0002      1      +   CUTs             CUT001   \n",
       "1     ST0003      1      +  ORF-T            YAL062W   \n",
       "2     ST3640      1      -   SUTs             SUT435   \n",
       "3     ST0004      1      +  ORF-T            YAL061W   \n",
       "4     ST0005      1      +  ORF-T            YAL060W   \n",
       "...      ...    ...    ...    ...                ...   \n",
       "4940  ST7262     16      -  ORF-T            YPR188C   \n",
       "4941  ST3625     16      +  ORF-T            YPR192W   \n",
       "4942  ST3627     16      +   SUTs             SUT429   \n",
       "4943  ST3629     16      +  other  YPR196W, YPRWtau4   \n",
       "4944  ST3633     16      +  ORF-T            YPR201W   \n",
       "\n",
       "                       genname   start     end  \n",
       "0                      unknown   30073   30905  \n",
       "1      Glutamate DeHydrogenase   31153   32985  \n",
       "2                      unknown   33077   34381  \n",
       "3                          NaN   33361   34897  \n",
       "4     Butanediol DeHydrogenase   35097   36393  \n",
       "...                        ...     ...     ...  \n",
       "4940         Myo1p Light Chain  911823  912511  \n",
       "4941      AQuaporin from Yeast  921755  922827  \n",
       "4942                   unknown  927659  928459  \n",
       "4943                   unknown  931203  933179  \n",
       "4944     ARsenicals Resistance  939683  942347  \n",
       "\n",
       "[4945 rows x 8 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_df.to_csv('genes_df.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Loading Chromosomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1.fsa  11.fsa 13.fsa 15.fsa 2.fsa  4.fsa  6.fsa  8.fsa\r\n",
      "10.fsa 12.fsa 14.fsa 16.fsa 3.fsa  5.fsa  7.fsa  9.fsa\r\n"
     ]
    }
   ],
   "source": [
    "!ls chromosome/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">>tpg|BK006935.2| [organism=Saccharomyces cerevisiae S288c] [strain=S288c] [moltype=genomic] [chromosome=I] [note=R64-1-1]\r\n",
      "CCACACCACACCCACACACCCACACACCACACCACACACCACACCACACCCACACACACA\r\n",
      "CATCCTAACACTACCCTAACACAGCCCTAATCTAACCCTGGCCAACCTGTCTCTCAACTT\r\n",
      "ACCCTCCATTACCCTGCCTCCACTCGTTACCCTGTCCCATTCAACCATACCACTCCGAAC\r\n",
      "CACCATCCATCCCTCTACTTACTACCACTCACCCACCGTTACCCTCCAATTACCCATATC\r\n",
      "CAACCCACTGCCACTTACCCTACCATTACCCTACCATCCACCATGACCTACTCACCATAC\r\n",
      "TGTTCTTCTACCCACCATATTGAAACGCTAACAAATGATCGTAAATAACACACACGTGCT\r\n",
      "TACCCTACCACTTTATACCACCACCACATGCCATACTCACCCTCACTTGTATACTGATTT\r\n",
      "TACGTACGCACACGGATGCTACAGTATATACCATCTCAAACTTACCCTACTCTCAGATTC\r\n",
      "CACTTCACTCCATGGCCCATCTCTCACTGAATCAGTACCAAATGCACTCACATCATTATG\r\n"
     ]
    }
   ],
   "source": [
    "!head -n 10 chromosome/1.fsa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "chromosomes = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in range(1,17):\n",
    "    with open(f'chromosome/{i}.fsa','r') as chromosome_file:\n",
    "        lines = chromosome_file.read().splitlines()\n",
    "        chromosomes[i] = ''.join(lines[1:])\n",
    "        chromosomes[i] = chromosomes[i].replace(\"\\n\", \"\").replace(\" \", \"\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chromosome 1: length->230218\n",
      "Chromosome 2: length->813184\n",
      "Chromosome 3: length->316620\n",
      "Chromosome 4: length->1531933\n",
      "Chromosome 5: length->576874\n",
      "Chromosome 6: length->270161\n",
      "Chromosome 7: length->1090940\n",
      "Chromosome 8: length->562643\n",
      "Chromosome 9: length->439888\n",
      "Chromosome 10: length->745751\n",
      "Chromosome 11: length->666816\n",
      "Chromosome 12: length->1078177\n",
      "Chromosome 13: length->924431\n",
      "Chromosome 14: length->784333\n",
      "Chromosome 15: length->1091291\n",
      "Chromosome 16: length->948066\n"
     ]
    }
   ],
   "source": [
    "for key,value in chromosomes.items():\n",
    "    print(f'Chromosome {key}: length->{len(value)}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Selecting gene from its start and end index from chromosome\n",
    "- positions are inclusive and start at 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "def select_gene(row):\n",
    "    string = chromosomes[row['chrom']][row['start']-1:row['end']]\n",
    "    if row['strand'] == '+':\n",
    "        return string\n",
    "    if row['strand'] == '-':\n",
    "        return(string[::-1].replace(\"C\", \"X\").replace(\"G\", \"C\").replace(\"X\", \"G\").replace(\"A\", \"X\").replace(\"T\", \"A\").replace(\"X\", \"T\"))\n",
    "    if row['strand'] == 'both':\n",
    "        return string\n",
    "genes_df['sequence'] = genes_df.apply(select_gene,axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_df.to_csv('genes_with_sequence.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "genesws = pd.read_csv('genes_with_sequence.csv',index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>strand</th>\n",
       "      <th>type</th>\n",
       "      <th>name</th>\n",
       "      <th>genname</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ST0002</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>CUTs</td>\n",
       "      <td>CUT001</td>\n",
       "      <td>unknown</td>\n",
       "      <td>30073</td>\n",
       "      <td>30905</td>\n",
       "      <td>TGTATTTTCCACATAGAAAATTCGATTTTTTTTTTTCAATGCACCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ST0003</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>YAL062W</td>\n",
       "      <td>Glutamate DeHydrogenase</td>\n",
       "      <td>31153</td>\n",
       "      <td>32985</td>\n",
       "      <td>TGACAGATATTCTGCACTTAAAAACTAAAAATATTATACCAACTTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ST3640</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>SUTs</td>\n",
       "      <td>SUT435</td>\n",
       "      <td>unknown</td>\n",
       "      <td>33077</td>\n",
       "      <td>34381</td>\n",
       "      <td>TTCTTTCATGCAATGTGATGTCCATCGGAGAAAACTGTATCTTGTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ST0004</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>YAL061W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>33361</td>\n",
       "      <td>34897</td>\n",
       "      <td>CGTCAAGATATTTGAAAGTTAATAGACAGTTAACAATAATAACAAC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ST0005</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>YAL060W</td>\n",
       "      <td>Butanediol DeHydrogenase</td>\n",
       "      <td>35097</td>\n",
       "      <td>36393</td>\n",
       "      <td>ACGATTGACCAAGTCAGAAAAAAAAAAAAAAAGGAACTAAAAAAAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4940</th>\n",
       "      <td>ST7262</td>\n",
       "      <td>16</td>\n",
       "      <td>-</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>YPR188C</td>\n",
       "      <td>Myo1p Light Chain</td>\n",
       "      <td>911823</td>\n",
       "      <td>912511</td>\n",
       "      <td>CGAATCGTTTATTGAGCAAACTTGAAGTAGATATGGACCATAGTGA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4941</th>\n",
       "      <td>ST3625</td>\n",
       "      <td>16</td>\n",
       "      <td>+</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>YPR192W</td>\n",
       "      <td>AQuaporin from Yeast</td>\n",
       "      <td>921755</td>\n",
       "      <td>922827</td>\n",
       "      <td>TGTCCTTTTGATATTTTGTTTTTTTTTTCCTTCTCTTTTTTTATTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4942</th>\n",
       "      <td>ST3627</td>\n",
       "      <td>16</td>\n",
       "      <td>+</td>\n",
       "      <td>SUTs</td>\n",
       "      <td>SUT429</td>\n",
       "      <td>unknown</td>\n",
       "      <td>927659</td>\n",
       "      <td>928459</td>\n",
       "      <td>AGATAAGAAGTGTTGCAGTAATATTATGTGTCACACCAGCGTGGCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4943</th>\n",
       "      <td>ST3629</td>\n",
       "      <td>16</td>\n",
       "      <td>+</td>\n",
       "      <td>other</td>\n",
       "      <td>YPR196W, YPRWtau4</td>\n",
       "      <td>unknown</td>\n",
       "      <td>931203</td>\n",
       "      <td>933179</td>\n",
       "      <td>TCCGGATGACTGCTTGCGTTACTATATATTTTTTGCTGTAATTCAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4944</th>\n",
       "      <td>ST3633</td>\n",
       "      <td>16</td>\n",
       "      <td>+</td>\n",
       "      <td>ORF-T</td>\n",
       "      <td>YPR201W</td>\n",
       "      <td>ARsenicals Resistance</td>\n",
       "      <td>939683</td>\n",
       "      <td>942347</td>\n",
       "      <td>GATTGTCAAGTTTTCCTCAATATTAGTTTGTTATACATTAGTTTTT...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4945 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          id  chrom strand   type               name  \\\n",
       "0     ST0002      1      +   CUTs             CUT001   \n",
       "1     ST0003      1      +  ORF-T            YAL062W   \n",
       "2     ST3640      1      -   SUTs             SUT435   \n",
       "3     ST0004      1      +  ORF-T            YAL061W   \n",
       "4     ST0005      1      +  ORF-T            YAL060W   \n",
       "...      ...    ...    ...    ...                ...   \n",
       "4940  ST7262     16      -  ORF-T            YPR188C   \n",
       "4941  ST3625     16      +  ORF-T            YPR192W   \n",
       "4942  ST3627     16      +   SUTs             SUT429   \n",
       "4943  ST3629     16      +  other  YPR196W, YPRWtau4   \n",
       "4944  ST3633     16      +  ORF-T            YPR201W   \n",
       "\n",
       "                       genname   start     end  \\\n",
       "0                      unknown   30073   30905   \n",
       "1      Glutamate DeHydrogenase   31153   32985   \n",
       "2                      unknown   33077   34381   \n",
       "3                          NaN   33361   34897   \n",
       "4     Butanediol DeHydrogenase   35097   36393   \n",
       "...                        ...     ...     ...   \n",
       "4940         Myo1p Light Chain  911823  912511   \n",
       "4941      AQuaporin from Yeast  921755  922827   \n",
       "4942                   unknown  927659  928459   \n",
       "4943                   unknown  931203  933179   \n",
       "4944     ARsenicals Resistance  939683  942347   \n",
       "\n",
       "                                               sequence  \n",
       "0     TGTATTTTCCACATAGAAAATTCGATTTTTTTTTTTCAATGCACCA...  \n",
       "1     TGACAGATATTCTGCACTTAAAAACTAAAAATATTATACCAACTTT...  \n",
       "2     TTCTTTCATGCAATGTGATGTCCATCGGAGAAAACTGTATCTTGTG...  \n",
       "3     CGTCAAGATATTTGAAAGTTAATAGACAGTTAACAATAATAACAAC...  \n",
       "4     ACGATTGACCAAGTCAGAAAAAAAAAAAAAAAGGAACTAAAAAAAG...  \n",
       "...                                                 ...  \n",
       "4940  CGAATCGTTTATTGAGCAAACTTGAAGTAGATATGGACCATAGTGA...  \n",
       "4941  TGTCCTTTTGATATTTTGTTTTTTTTTTCCTTCTCTTTTTTTATTC...  \n",
       "4942  AGATAAGAAGTGTTGCAGTAATATTATGTGTCACACCAGCGTGGCT...  \n",
       "4943  TCCGGATGACTGCTTGCGTTACTATATATTTTTTGCTGTAATTCAG...  \n",
       "4944  GATTGTCAAGTTTTCCTCAATATTAGTTTGTTATACATTAGTTTTT...  \n",
       "\n",
       "[4945 rows x 9 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genesws"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "id          4945\n",
       "chrom         16\n",
       "strand         2\n",
       "type           4\n",
       "name        4921\n",
       "genname     1658\n",
       "start       4931\n",
       "end         4932\n",
       "sequence    4945\n",
       "dtype: int64"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genesws.nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['CUTs', 'ORF-T', 'SUTs', 'other'], dtype=object)"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genesws.type.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "cut_df = genesws[genesws['type'] == 'CUTs'].reset_index(drop=True)\n",
    "cut_df.to_csv(\"cut_table.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>strand</th>\n",
       "      <th>name</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>CUT001</td>\n",
       "      <td>30073</td>\n",
       "      <td>30905</td>\n",
       "      <td>TGTATTTTCCACATAGAAAATTCGATTTTTTTTTTTCAATGCACCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>CUT003</td>\n",
       "      <td>143439</td>\n",
       "      <td>143599</td>\n",
       "      <td>CTTGTCGAAAATACGCGGTGTAGGGAGTTATGGTGGATAACTTTTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>CUT004</td>\n",
       "      <td>170495</td>\n",
       "      <td>172447</td>\n",
       "      <td>AATTACTTCCCTGGCTCGCTCCTCCACTGCCTGGGTAAATTGTTCC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>+</td>\n",
       "      <td>CUT006</td>\n",
       "      <td>96354</td>\n",
       "      <td>96570</td>\n",
       "      <td>ATTTTGGTAAATGCACACGGATAAATGCATGATATATAAAATATTA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>-</td>\n",
       "      <td>CUT445</td>\n",
       "      <td>278428</td>\n",
       "      <td>280876</td>\n",
       "      <td>AATGGTAGTCATATCATGTCAAGAATAGGTATCCAAAACGCAGCGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>838</th>\n",
       "      <td>16</td>\n",
       "      <td>+</td>\n",
       "      <td>CUT433</td>\n",
       "      <td>822051</td>\n",
       "      <td>822163</td>\n",
       "      <td>AAACTTGCCGTTAACAATAAGTGACCTTGCATGGAAGACTGTCGAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>839</th>\n",
       "      <td>16</td>\n",
       "      <td>-</td>\n",
       "      <td>CUT918</td>\n",
       "      <td>834071</td>\n",
       "      <td>834303</td>\n",
       "      <td>TGCTAACTTTTATCATCAAATCAGTAAACCCCTTCTTGTTTTTTCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>840</th>\n",
       "      <td>16</td>\n",
       "      <td>-</td>\n",
       "      <td>CUT919</td>\n",
       "      <td>874527</td>\n",
       "      <td>874879</td>\n",
       "      <td>CACAGATGAGGTCAGATTTAACTATTGACCTTCTTGAAACCAAATA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>841</th>\n",
       "      <td>16</td>\n",
       "      <td>-</td>\n",
       "      <td>CUT921</td>\n",
       "      <td>878151</td>\n",
       "      <td>878495</td>\n",
       "      <td>AACAGGTTCTTTCAACTCTGGAAATTCATCCACAATCTTGTCAGCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>842</th>\n",
       "      <td>16</td>\n",
       "      <td>-</td>\n",
       "      <td>CUT923</td>\n",
       "      <td>901103</td>\n",
       "      <td>901543</td>\n",
       "      <td>GTATAGCTGGTAGCAAACGTAGAAGAACAGAATGGACCAGAAAGTA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>843 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     chrom strand    name   start     end  \\\n",
       "0        1      +  CUT001   30073   30905   \n",
       "1        1      +  CUT003  143439  143599   \n",
       "2        1      +  CUT004  170495  172447   \n",
       "3        2      +  CUT006   96354   96570   \n",
       "4        2      -  CUT445  278428  280876   \n",
       "..     ...    ...     ...     ...     ...   \n",
       "838     16      +  CUT433  822051  822163   \n",
       "839     16      -  CUT918  834071  834303   \n",
       "840     16      -  CUT919  874527  874879   \n",
       "841     16      -  CUT921  878151  878495   \n",
       "842     16      -  CUT923  901103  901543   \n",
       "\n",
       "                                              sequence  \n",
       "0    TGTATTTTCCACATAGAAAATTCGATTTTTTTTTTTCAATGCACCA...  \n",
       "1    CTTGTCGAAAATACGCGGTGTAGGGAGTTATGGTGGATAACTTTTT...  \n",
       "2    AATTACTTCCCTGGCTCGCTCCTCCACTGCCTGGGTAAATTGTTCC...  \n",
       "3    ATTTTGGTAAATGCACACGGATAAATGCATGATATATAAAATATTA...  \n",
       "4    AATGGTAGTCATATCATGTCAAGAATAGGTATCCAAAACGCAGCGG...  \n",
       "..                                                 ...  \n",
       "838  AAACTTGCCGTTAACAATAAGTGACCTTGCATGGAAGACTGTCGAT...  \n",
       "839  TGCTAACTTTTATCATCAAATCAGTAAACCCCTTCTTGTTTTTTCT...  \n",
       "840  CACAGATGAGGTCAGATTTAACTATTGACCTTCTTGAAACCAAATA...  \n",
       "841  AACAGGTTCTTTCAACTCTGGAAATTCATCCACAATCTTGTCAGCA...  \n",
       "842  GTATAGCTGGTAGCAAACGTAGAAGAACAGAATGGACCAGAAAGTA...  \n",
       "\n",
       "[843 rows x 6 columns]"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cut_df_needed_samina = cut_df.drop(columns=['type', 'genname', 'id'])\n",
    "cut_df_needed_samina"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "chrom        16\n",
       "strand        2\n",
       "name        843\n",
       "start       843\n",
       "end         842\n",
       "sequence    843\n",
       "dtype: int64"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cut_df_needed_samina.nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "cut_df_needed_samina.to_csv(\"overlapping_CUTs.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "noncoding_df = pd.read_csv('noncoding_genes.csv', header = None)\n",
    "noncoding_df.columns = ['chrom', 'strand', 'name', 'start', 'end', 'sequence']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "chrom        17\n",
       "strand        2\n",
       "name         30\n",
       "start       435\n",
       "end         433\n",
       "sequence    219\n",
       "dtype: int64"
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "noncoding_df.nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TGTATTTTCCACATAGAAAATTCGATTTTTTTTTTTCAATGCACCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CTTGTCGAAAATACGCGGTGTAGGGAGTTATGGTGGATAACTTTTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AATTACTTCCCTGGCTCGCTCCTCCACTGCCTGGGTAAATTGTTCC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ATTTTGGTAAATGCACACGGATAAATGCATGATATATAAAATATTA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AATGGTAGTCATATCATGTCAAGAATAGGTATCCAAAACGCAGCGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>838</th>\n",
       "      <td>AAACTTGCCGTTAACAATAAGTGACCTTGCATGGAAGACTGTCGAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>839</th>\n",
       "      <td>TGCTAACTTTTATCATCAAATCAGTAAACCCCTTCTTGTTTTTTCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>840</th>\n",
       "      <td>CACAGATGAGGTCAGATTTAACTATTGACCTTCTTGAAACCAAATA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>841</th>\n",
       "      <td>AACAGGTTCTTTCAACTCTGGAAATTCATCCACAATCTTGTCAGCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>842</th>\n",
       "      <td>GTATAGCTGGTAGCAAACGTAGAAGAACAGAATGGACCAGAAAGTA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>843 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              sequence\n",
       "0    TGTATTTTCCACATAGAAAATTCGATTTTTTTTTTTCAATGCACCA...\n",
       "1    CTTGTCGAAAATACGCGGTGTAGGGAGTTATGGTGGATAACTTTTT...\n",
       "2    AATTACTTCCCTGGCTCGCTCCTCCACTGCCTGGGTAAATTGTTCC...\n",
       "3    ATTTTGGTAAATGCACACGGATAAATGCATGATATATAAAATATTA...\n",
       "4    AATGGTAGTCATATCATGTCAAGAATAGGTATCCAAAACGCAGCGG...\n",
       "..                                                 ...\n",
       "838  AAACTTGCCGTTAACAATAAGTGACCTTGCATGGAAGACTGTCGAT...\n",
       "839  TGCTAACTTTTATCATCAAATCAGTAAACCCCTTCTTGTTTTTTCT...\n",
       "840  CACAGATGAGGTCAGATTTAACTATTGACCTTCTTGAAACCAAATA...\n",
       "841  AACAGGTTCTTTCAACTCTGGAAATTCATCCACAATCTTGTCAGCA...\n",
       "842  GTATAGCTGGTAGCAAACGTAGAAGAACAGAATGGACCAGAAAGTA...\n",
       "\n",
       "[843 rows x 1 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "overlapping_cut_sequences = cut_df_needed_samina.drop(columns=['chrom', 'strand', 'name', 'start','end'])\n",
    "overlapping_cut_sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GGGCCCTTTCTTCCGTTTGAACGTAAAGGCATTTTTGAGACCATTA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GTGAATGATGAATTTAATTCTTTGGTCCGTGTTTATGATGGGAAGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GGTTGTTTGGCCGAGCGGTCTAAGGCGCCTGATTCAAGAAATATCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GGGCGTGTGGTCTAGTGGTATGATTCTCGCTTTGGGCGACTTCCTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GGGCACATGGCGCAGTTGGTAGCGCGCTTCCCTTGCAAGGAAGAGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>GCGGATTTAGCTCAGTTGGGAGAGCGCCAGACTGAAGAAAAAACTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>433</th>\n",
       "      <td>GCGGATTTAGCTCAGTTGGGAGAGCGCCAGACTGAAGAAAAAACTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>434</th>\n",
       "      <td>TCCTTGTTAGCTCAGTTGGTAGAGCGTTCGGCTTTTAAGCGCATTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>435</th>\n",
       "      <td>GCTCGTATGGCGCAGTGGTAGCGCAGCAGATTGCAAATCTGTTGGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>436</th>\n",
       "      <td>GGTCTCTTGGCCCAGTTGGTTAAGGCACCGTGCTAATAACGCGGGG...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>437 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              sequence\n",
       "0    GGGCCCTTTCTTCCGTTTGAACGTAAAGGCATTTTTGAGACCATTA...\n",
       "1    GTGAATGATGAATTTAATTCTTTGGTCCGTGTTTATGATGGGAAGT...\n",
       "2    GGTTGTTTGGCCGAGCGGTCTAAGGCGCCTGATTCAAGAAATATCT...\n",
       "3    GGGCGTGTGGTCTAGTGGTATGATTCTCGCTTTGGGCGACTTCCTG...\n",
       "4    GGGCACATGGCGCAGTTGGTAGCGCGCTTCCCTTGCAAGGAAGAGG...\n",
       "..                                                 ...\n",
       "432  GCGGATTTAGCTCAGTTGGGAGAGCGCCAGACTGAAGAAAAAACTT...\n",
       "433  GCGGATTTAGCTCAGTTGGGAGAGCGCCAGACTGAAGAAAAAACTT...\n",
       "434  TCCTTGTTAGCTCAGTTGGTAGAGCGTTCGGCTTTTAAGCGCATTT...\n",
       "435  GCTCGTATGGCGCAGTGGTAGCGCAGCAGATTGCAAATCTGTTGGT...\n",
       "436  GGTCTCTTGGCCCAGTTGGTTAAGGCACCGTGCTAATAACGCGGGG...\n",
       "\n",
       "[437 rows x 1 columns]"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "noncoding_sequences = noncoding_df.drop(columns=['chrom', 'strand', 'name', 'start','end'])\n",
    "noncoding_sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "nonoverlapping_cuts_df = pd.read_csv('nonoverlapping_CUTs.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Type</th>\n",
       "      <th>ID</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Transcribed_strand</th>\n",
       "      <th>Deletion_start_site</th>\n",
       "      <th>Deletion_end_site</th>\n",
       "      <th>YP 30C+2% Glucose</th>\n",
       "      <th>YP 30C+2% Glucose+5% Methanol</th>\n",
       "      <th>YP 30C+2% Glucose+0.4% TCA</th>\n",
       "      <th>YP 30C+2% Galactose</th>\n",
       "      <th>...</th>\n",
       "      <th>LSS-ESS_C-limited 30C_q.value</th>\n",
       "      <th>LSS-ESS_C-limited 30C_FC</th>\n",
       "      <th>LSS-ESS_N-limited 36C_q.value</th>\n",
       "      <th>LSS-ESS_N-limited 36C_FC</th>\n",
       "      <th>LSS-ESS_C-limited 36C_q.value</th>\n",
       "      <th>LSS-ESS_C-limited 36C_FC</th>\n",
       "      <th>LSS-ESS_N-limited 30C + 100mM LiCl_q.value</th>\n",
       "      <th>LSS-ESS_N-limited 30C + 100mM LiCl_FC</th>\n",
       "      <th>LSS-ESS_C-limited 30C + 100mM LiCl_q.value</th>\n",
       "      <th>LSS-ESS_C-limited 30C + 100mM LiCl_FC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CUT</td>\n",
       "      <td>CUT001--CUT437</td>\n",
       "      <td>1</td>\n",
       "      <td>both</td>\n",
       "      <td>30028</td>\n",
       "      <td>30950</td>\n",
       "      <td>-0.128454283</td>\n",
       "      <td>non-significant</td>\n",
       "      <td>non-significant</td>\n",
       "      <td>non-significant</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.966955</td>\n",
       "      <td>2.390000e-04</td>\n",
       "      <td>0.020278</td>\n",
       "      <td>0.389297</td>\n",
       "      <td>0.256269</td>\n",
       "      <td>0.293861</td>\n",
       "      <td>0.058330</td>\n",
       "      <td>0.001968</td>\n",
       "      <td>0.023032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CUT</td>\n",
       "      <td>CUT002</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>138607</td>\n",
       "      <td>138831</td>\n",
       "      <td>-0.644129986</td>\n",
       "      <td>-0.805297093</td>\n",
       "      <td>-0.216548183</td>\n",
       "      <td>-0.320866634</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>15.007987</td>\n",
       "      <td>7.280077e-03</td>\n",
       "      <td>0.019060</td>\n",
       "      <td>0.191974</td>\n",
       "      <td>0.284571</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.157434</td>\n",
       "      <td>0.579547</td>\n",
       "      <td>0.061475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CUT</td>\n",
       "      <td>CUT005</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>180255</td>\n",
       "      <td>180511</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.335316</td>\n",
       "      <td>1.061972e-03</td>\n",
       "      <td>0.031227</td>\n",
       "      <td>0.027180</td>\n",
       "      <td>0.051301</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.278810</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.773978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CUT</td>\n",
       "      <td>CUT007</td>\n",
       "      <td>2</td>\n",
       "      <td>+</td>\n",
       "      <td>191991</td>\n",
       "      <td>192343</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.583513</td>\n",
       "      <td>3.590000e-07</td>\n",
       "      <td>0.003830</td>\n",
       "      <td>0.152099</td>\n",
       "      <td>0.182194</td>\n",
       "      <td>0.143802</td>\n",
       "      <td>0.065291</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.099942</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CUT</td>\n",
       "      <td>CUT008</td>\n",
       "      <td>2</td>\n",
       "      <td>+</td>\n",
       "      <td>237744</td>\n",
       "      <td>237928</td>\n",
       "      <td>non-significant</td>\n",
       "      <td>non-significant</td>\n",
       "      <td>-0.381190311</td>\n",
       "      <td>non-significant</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>12.377517</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.139347</td>\n",
       "      <td>0.944729</td>\n",
       "      <td>0.178498</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.097792</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.537298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>CUT</td>\n",
       "      <td>CUT873</td>\n",
       "      <td>15</td>\n",
       "      <td>-</td>\n",
       "      <td>1003415</td>\n",
       "      <td>1004087</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.269072</td>\n",
       "      <td>6.270000e-04</td>\n",
       "      <td>0.016518</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.324277</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>4.442078</td>\n",
       "      <td>0.124352</td>\n",
       "      <td>0.036079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>SUT-CUT</td>\n",
       "      <td>SUT086--CUT102</td>\n",
       "      <td>5</td>\n",
       "      <td>+</td>\n",
       "      <td>7668</td>\n",
       "      <td>8134</td>\n",
       "      <td>-0.272163352</td>\n",
       "      <td>-0.188385856</td>\n",
       "      <td>-0.377701506</td>\n",
       "      <td>-0.481597931</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.896995</td>\n",
       "      <td>3.557592e-03</td>\n",
       "      <td>0.014247</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.858757</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.118071</td>\n",
       "      <td>0.816510</td>\n",
       "      <td>0.065340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>SUT-CUT</td>\n",
       "      <td>SUT233--CUT707</td>\n",
       "      <td>11</td>\n",
       "      <td>both</td>\n",
       "      <td>229966</td>\n",
       "      <td>231479</td>\n",
       "      <td>-0.716207371</td>\n",
       "      <td>-0.644378768</td>\n",
       "      <td>-0.856118859</td>\n",
       "      <td>-0.839265734</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8.447845</td>\n",
       "      <td>1.797692e-02</td>\n",
       "      <td>0.019331</td>\n",
       "      <td>0.057661</td>\n",
       "      <td>0.041079</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.086992</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.563029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>SUT-CUT</td>\n",
       "      <td>SUT420--CUT425</td>\n",
       "      <td>16</td>\n",
       "      <td>+</td>\n",
       "      <td>607502</td>\n",
       "      <td>608688</td>\n",
       "      <td>non-significant</td>\n",
       "      <td>non-significant</td>\n",
       "      <td>non-significant</td>\n",
       "      <td>non-significant</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.806966</td>\n",
       "      <td>2.340000e-04</td>\n",
       "      <td>0.006599</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.291476</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.121357</td>\n",
       "      <td>0.884281</td>\n",
       "      <td>0.059028</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>SUT-CUT</td>\n",
       "      <td>SUT473--CUT502</td>\n",
       "      <td>4</td>\n",
       "      <td>-</td>\n",
       "      <td>621019</td>\n",
       "      <td>621693</td>\n",
       "      <td>non-significant</td>\n",
       "      <td>non-significant</td>\n",
       "      <td>non-significant</td>\n",
       "      <td>non-significant</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.645708</td>\n",
       "      <td>8.796011e-03</td>\n",
       "      <td>0.025489</td>\n",
       "      <td>0.935535</td>\n",
       "      <td>0.311145</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.150809</td>\n",
       "      <td>0.002728</td>\n",
       "      <td>0.023622</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>67 rows × 50 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Type              ID  Chromosome Transcribed_strand  \\\n",
       "0       CUT  CUT001--CUT437           1               both   \n",
       "1       CUT          CUT002           1                  +   \n",
       "2       CUT          CUT005           1                  +   \n",
       "3       CUT          CUT007           2                  +   \n",
       "4       CUT          CUT008           2                  +   \n",
       "..      ...             ...         ...                ...   \n",
       "62      CUT          CUT873          15                  -   \n",
       "63  SUT-CUT  SUT086--CUT102           5                  +   \n",
       "64  SUT-CUT  SUT233--CUT707          11               both   \n",
       "65  SUT-CUT  SUT420--CUT425          16                  +   \n",
       "66  SUT-CUT  SUT473--CUT502           4                  -   \n",
       "\n",
       "    Deletion_start_site  Deletion_end_site YP 30C+2% Glucose  \\\n",
       "0                 30028              30950      -0.128454283   \n",
       "1                138607             138831      -0.644129986   \n",
       "2                180255             180511               NaN   \n",
       "3                191991             192343               NaN   \n",
       "4                237744             237928   non-significant   \n",
       "..                  ...                ...               ...   \n",
       "62              1003415            1004087               NaN   \n",
       "63                 7668               8134      -0.272163352   \n",
       "64               229966             231479      -0.716207371   \n",
       "65               607502             608688   non-significant   \n",
       "66               621019             621693   non-significant   \n",
       "\n",
       "   YP 30C+2% Glucose+5% Methanol YP 30C+2% Glucose+0.4% TCA  \\\n",
       "0                non-significant            non-significant   \n",
       "1                   -0.805297093               -0.216548183   \n",
       "2                            NaN                        NaN   \n",
       "3                            NaN                        NaN   \n",
       "4                non-significant               -0.381190311   \n",
       "..                           ...                        ...   \n",
       "62                           NaN                        NaN   \n",
       "63                  -0.188385856               -0.377701506   \n",
       "64                  -0.644378768               -0.856118859   \n",
       "65               non-significant            non-significant   \n",
       "66               non-significant            non-significant   \n",
       "\n",
       "   YP 30C+2% Galactose  ... LSS-ESS_C-limited 30C_q.value  \\\n",
       "0      non-significant  ...                           1.0   \n",
       "1         -0.320866634  ...                           1.0   \n",
       "2                  NaN  ...                           1.0   \n",
       "3                  NaN  ...                           1.0   \n",
       "4      non-significant  ...                           1.0   \n",
       "..                 ...  ...                           ...   \n",
       "62                 NaN  ...                           1.0   \n",
       "63        -0.481597931  ...                           1.0   \n",
       "64        -0.839265734  ...                           1.0   \n",
       "65     non-significant  ...                           1.0   \n",
       "66     non-significant  ...                           1.0   \n",
       "\n",
       "   LSS-ESS_C-limited 30C_FC LSS-ESS_N-limited 36C_q.value  \\\n",
       "0                  1.966955                  2.390000e-04   \n",
       "1                 15.007987                  7.280077e-03   \n",
       "2                  5.335316                  1.061972e-03   \n",
       "3                  3.583513                  3.590000e-07   \n",
       "4                 12.377517                  1.000000e+00   \n",
       "..                      ...                           ...   \n",
       "62                 0.269072                  6.270000e-04   \n",
       "63                 1.896995                  3.557592e-03   \n",
       "64                 8.447845                  1.797692e-02   \n",
       "65                 1.806966                  2.340000e-04   \n",
       "66                 1.645708                  8.796011e-03   \n",
       "\n",
       "   LSS-ESS_N-limited 36C_FC LSS-ESS_C-limited 36C_q.value  \\\n",
       "0                  0.020278                      0.389297   \n",
       "1                  0.019060                      0.191974   \n",
       "2                  0.031227                      0.027180   \n",
       "3                  0.003830                      0.152099   \n",
       "4                  0.139347                      0.944729   \n",
       "..                      ...                           ...   \n",
       "62                 0.016518                      1.000000   \n",
       "63                 0.014247                      1.000000   \n",
       "64                 0.019331                      0.057661   \n",
       "65                 0.006599                      1.000000   \n",
       "66                 0.025489                      0.935535   \n",
       "\n",
       "   LSS-ESS_C-limited 36C_FC LSS-ESS_N-limited 30C + 100mM LiCl_q.value  \\\n",
       "0                  0.256269                                   0.293861   \n",
       "1                  0.284571                                   1.000000   \n",
       "2                  0.051301                                   1.000000   \n",
       "3                  0.182194                                   0.143802   \n",
       "4                  0.178498                                   1.000000   \n",
       "..                      ...                                        ...   \n",
       "62                 0.324277                                   1.000000   \n",
       "63                 0.858757                                   1.000000   \n",
       "64                 0.041079                                   1.000000   \n",
       "65                 0.291476                                   1.000000   \n",
       "66                 0.311145                                   1.000000   \n",
       "\n",
       "   LSS-ESS_N-limited 30C + 100mM LiCl_FC  \\\n",
       "0                               0.058330   \n",
       "1                               0.157434   \n",
       "2                               0.278810   \n",
       "3                               0.065291   \n",
       "4                               0.097792   \n",
       "..                                   ...   \n",
       "62                              4.442078   \n",
       "63                              0.118071   \n",
       "64                              0.086992   \n",
       "65                              0.121357   \n",
       "66                              0.150809   \n",
       "\n",
       "   LSS-ESS_C-limited 30C + 100mM LiCl_q.value  \\\n",
       "0                                    0.001968   \n",
       "1                                    0.579547   \n",
       "2                                    1.000000   \n",
       "3                                    1.000000   \n",
       "4                                    1.000000   \n",
       "..                                        ...   \n",
       "62                                   0.124352   \n",
       "63                                   0.816510   \n",
       "64                                   1.000000   \n",
       "65                                   0.884281   \n",
       "66                                   0.002728   \n",
       "\n",
       "   LSS-ESS_C-limited 30C + 100mM LiCl_FC  \n",
       "0                               0.023032  \n",
       "1                               0.061475  \n",
       "2                               0.773978  \n",
       "3                               0.099942  \n",
       "4                               0.537298  \n",
       "..                                   ...  \n",
       "62                              0.036079  \n",
       "63                              0.065340  \n",
       "64                              0.563029  \n",
       "65                              0.059028  \n",
       "66                              0.023622  \n",
       "\n",
       "[67 rows x 50 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nonoverlapping_cuts_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "+       35\n",
       "-       26\n",
       "both     6\n",
       "Name: Transcribed_strand, dtype: int64"
      ]
     },
     "execution_count": 176,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nonoverlapping_cuts_df.Transcribed_strand.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "nonoverlapping_cuts = nonoverlapping_cuts_df[['Type', 'ID', 'Chromosome', \"Transcribed_strand\", \"Deletion_start_site\", \"Deletion_end_site\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CUT        59\n",
       "CUT-SUT     4\n",
       "SUT-CUT     4\n",
       "Name: Type, dtype: int64"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nonoverlapping_cuts.Type.value_counts()"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/anaconda3/lib/python3.8/site-packages/pandas/core/frame.py:4296: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  return super().rename(\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Type</th>\n",
       "      <th>ID</th>\n",
       "      <th>chrom</th>\n",
       "      <th>Transcribed_strand</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CUT</td>\n",
       "      <td>CUT001--CUT437</td>\n",
       "      <td>1</td>\n",
       "      <td>both</td>\n",
       "      <td>30028</td>\n",
       "      <td>30950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CUT</td>\n",
       "      <td>CUT002</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>138607</td>\n",
       "      <td>138831</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CUT</td>\n",
       "      <td>CUT005</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>180255</td>\n",
       "      <td>180511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CUT</td>\n",
       "      <td>CUT007</td>\n",
       "      <td>2</td>\n",
       "      <td>+</td>\n",
       "      <td>191991</td>\n",
       "      <td>192343</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CUT</td>\n",
       "      <td>CUT008</td>\n",
       "      <td>2</td>\n",
       "      <td>+</td>\n",
       "      <td>237744</td>\n",
       "      <td>237928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>CUT</td>\n",
       "      <td>CUT873</td>\n",
       "      <td>15</td>\n",
       "      <td>-</td>\n",
       "      <td>1003415</td>\n",
       "      <td>1004087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>SUT-CUT</td>\n",
       "      <td>SUT086--CUT102</td>\n",
       "      <td>5</td>\n",
       "      <td>+</td>\n",
       "      <td>7668</td>\n",
       "      <td>8134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>SUT-CUT</td>\n",
       "      <td>SUT233--CUT707</td>\n",
       "      <td>11</td>\n",
       "      <td>both</td>\n",
       "      <td>229966</td>\n",
       "      <td>231479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>SUT-CUT</td>\n",
       "      <td>SUT420--CUT425</td>\n",
       "      <td>16</td>\n",
       "      <td>+</td>\n",
       "      <td>607502</td>\n",
       "      <td>608688</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>SUT-CUT</td>\n",
       "      <td>SUT473--CUT502</td>\n",
       "      <td>4</td>\n",
       "      <td>-</td>\n",
       "      <td>621019</td>\n",
       "      <td>621693</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>67 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Type              ID  chrom Transcribed_strand    start      end\n",
       "0       CUT  CUT001--CUT437      1               both    30028    30950\n",
       "1       CUT          CUT002      1                  +   138607   138831\n",
       "2       CUT          CUT005      1                  +   180255   180511\n",
       "3       CUT          CUT007      2                  +   191991   192343\n",
       "4       CUT          CUT008      2                  +   237744   237928\n",
       "..      ...             ...    ...                ...      ...      ...\n",
       "62      CUT          CUT873     15                  -  1003415  1004087\n",
       "63  SUT-CUT  SUT086--CUT102      5                  +     7668     8134\n",
       "64  SUT-CUT  SUT233--CUT707     11               both   229966   231479\n",
       "65  SUT-CUT  SUT420--CUT425     16                  +   607502   608688\n",
       "66  SUT-CUT  SUT473--CUT502      4                  -   621019   621693\n",
       "\n",
       "[67 rows x 6 columns]"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nonoverlapping_cuts.rename(columns={'Deletion_start_site' : 'start', 'Deletion_end_site' : 'end', 'Chromosome' : 'chrom'}, inplace = True)\n",
    "nonoverlapping_cuts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-104-4a649fd4d23d>:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  nonoverlapping_cuts['sequence'] = genes_df.apply(select_gene,axis=1)\n"
     ]
    }
   ],
   "source": [
    "nonoverlapping_cuts['sequence'] = genes_df.apply(select_gene,axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Type</th>\n",
       "      <th>ID</th>\n",
       "      <th>chrom</th>\n",
       "      <th>Transcribed_strand</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CUT</td>\n",
       "      <td>CUT001--CUT437</td>\n",
       "      <td>1</td>\n",
       "      <td>both</td>\n",
       "      <td>30028</td>\n",
       "      <td>30950</td>\n",
       "      <td>TGTATTTTCCACATAGAAAATTCGATTTTTTTTTTTCAATGCACCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CUT</td>\n",
       "      <td>CUT002</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>138607</td>\n",
       "      <td>138831</td>\n",
       "      <td>TGACAGATATTCTGCACTTAAAAACTAAAAATATTATACCAACTTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CUT</td>\n",
       "      <td>CUT005</td>\n",
       "      <td>1</td>\n",
       "      <td>+</td>\n",
       "      <td>180255</td>\n",
       "      <td>180511</td>\n",
       "      <td>TTCTTTCATGCAATGTGATGTCCATCGGAGAAAACTGTATCTTGTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CUT</td>\n",
       "      <td>CUT007</td>\n",
       "      <td>2</td>\n",
       "      <td>+</td>\n",
       "      <td>191991</td>\n",
       "      <td>192343</td>\n",
       "      <td>CGTCAAGATATTTGAAAGTTAATAGACAGTTAACAATAATAACAAC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CUT</td>\n",
       "      <td>CUT008</td>\n",
       "      <td>2</td>\n",
       "      <td>+</td>\n",
       "      <td>237744</td>\n",
       "      <td>237928</td>\n",
       "      <td>ACGATTGACCAAGTCAGAAAAAAAAAAAAAAAGGAACTAAAAAAAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>CUT</td>\n",
       "      <td>CUT873</td>\n",
       "      <td>15</td>\n",
       "      <td>-</td>\n",
       "      <td>1003415</td>\n",
       "      <td>1004087</td>\n",
       "      <td>GTGACGACGATATACAGATCCTATATTCTTCATGTATGTAATAGAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>SUT-CUT</td>\n",
       "      <td>SUT086--CUT102</td>\n",
       "      <td>5</td>\n",
       "      <td>+</td>\n",
       "      <td>7668</td>\n",
       "      <td>8134</td>\n",
       "      <td>GCAGTAAAAATAGGGTAAACACAGGTAAAAGACGGTATGGAAACGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>SUT-CUT</td>\n",
       "      <td>SUT233--CUT707</td>\n",
       "      <td>11</td>\n",
       "      <td>both</td>\n",
       "      <td>229966</td>\n",
       "      <td>231479</td>\n",
       "      <td>GAAAAAAATGCAGACCTTTCTTAATTCAACTTTTTTTCTTTCAAAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>SUT-CUT</td>\n",
       "      <td>SUT420--CUT425</td>\n",
       "      <td>16</td>\n",
       "      <td>+</td>\n",
       "      <td>607502</td>\n",
       "      <td>608688</td>\n",
       "      <td>TTGAGTACAGTGGGACGATGTTTCAAAGGTCTGGCGCTGCTCATCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>SUT-CUT</td>\n",
       "      <td>SUT473--CUT502</td>\n",
       "      <td>4</td>\n",
       "      <td>-</td>\n",
       "      <td>621019</td>\n",
       "      <td>621693</td>\n",
       "      <td>GGTGGTGATAAAGGGAAAATGCGCGCGGGATTAGCAAAGGCGGCCA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>67 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Type              ID  chrom Transcribed_strand    start      end  \\\n",
       "0       CUT  CUT001--CUT437      1               both    30028    30950   \n",
       "1       CUT          CUT002      1                  +   138607   138831   \n",
       "2       CUT          CUT005      1                  +   180255   180511   \n",
       "3       CUT          CUT007      2                  +   191991   192343   \n",
       "4       CUT          CUT008      2                  +   237744   237928   \n",
       "..      ...             ...    ...                ...      ...      ...   \n",
       "62      CUT          CUT873     15                  -  1003415  1004087   \n",
       "63  SUT-CUT  SUT086--CUT102      5                  +     7668     8134   \n",
       "64  SUT-CUT  SUT233--CUT707     11               both   229966   231479   \n",
       "65  SUT-CUT  SUT420--CUT425     16                  +   607502   608688   \n",
       "66  SUT-CUT  SUT473--CUT502      4                  -   621019   621693   \n",
       "\n",
       "                                             sequence  \n",
       "0   TGTATTTTCCACATAGAAAATTCGATTTTTTTTTTTCAATGCACCA...  \n",
       "1   TGACAGATATTCTGCACTTAAAAACTAAAAATATTATACCAACTTT...  \n",
       "2   TTCTTTCATGCAATGTGATGTCCATCGGAGAAAACTGTATCTTGTG...  \n",
       "3   CGTCAAGATATTTGAAAGTTAATAGACAGTTAACAATAATAACAAC...  \n",
       "4   ACGATTGACCAAGTCAGAAAAAAAAAAAAAAAGGAACTAAAAAAAG...  \n",
       "..                                                ...  \n",
       "62  GTGACGACGATATACAGATCCTATATTCTTCATGTATGTAATAGAT...  \n",
       "63  GCAGTAAAAATAGGGTAAACACAGGTAAAAGACGGTATGGAAACGG...  \n",
       "64  GAAAAAAATGCAGACCTTTCTTAATTCAACTTTTTTTCTTTCAAAT...  \n",
       "65  TTGAGTACAGTGGGACGATGTTTCAAAGGTCTGGCGCTGCTCATCA...  \n",
       "66  GGTGGTGATAAAGGGAAAATGCGCGCGGGATTAGCAAAGGCGGCCA...  \n",
       "\n",
       "[67 rows x 7 columns]"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nonoverlapping_cuts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [],
   "source": [
    "nonoverlapping_cuts.to_csv(\"non_overlapping_CUTs.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Type                   3\n",
       "ID                    67\n",
       "chrom                 16\n",
       "Transcribed_strand     3\n",
       "start                 67\n",
       "end                   67\n",
       "sequence              67\n",
       "dtype: int64"
      ]
     },
     "execution_count": 175,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nonoverlapping_cuts.nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [],
   "source": [
    "nonoverlapping_cut_sequences = nonoverlapping_cuts.drop(columns=['Type', 'ID', 'chrom', 'Transcribed_strand','end', 'start'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TGTATTTTCCACATAGAAAATTCGATTTTTTTTTTTCAATGCACCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TGACAGATATTCTGCACTTAAAAACTAAAAATATTATACCAACTTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TTCTTTCATGCAATGTGATGTCCATCGGAGAAAACTGTATCTTGTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CGTCAAGATATTTGAAAGTTAATAGACAGTTAACAATAATAACAAC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACGATTGACCAAGTCAGAAAAAAAAAAAAAAAGGAACTAAAAAAAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>GTGACGACGATATACAGATCCTATATTCTTCATGTATGTAATAGAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>GCAGTAAAAATAGGGTAAACACAGGTAAAAGACGGTATGGAAACGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>GAAAAAAATGCAGACCTTTCTTAATTCAACTTTTTTTCTTTCAAAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>TTGAGTACAGTGGGACGATGTTTCAAAGGTCTGGCGCTGCTCATCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>GGTGGTGATAAAGGGAAAATGCGCGCGGGATTAGCAAAGGCGGCCA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>67 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             sequence\n",
       "0   TGTATTTTCCACATAGAAAATTCGATTTTTTTTTTTCAATGCACCA...\n",
       "1   TGACAGATATTCTGCACTTAAAAACTAAAAATATTATACCAACTTT...\n",
       "2   TTCTTTCATGCAATGTGATGTCCATCGGAGAAAACTGTATCTTGTG...\n",
       "3   CGTCAAGATATTTGAAAGTTAATAGACAGTTAACAATAATAACAAC...\n",
       "4   ACGATTGACCAAGTCAGAAAAAAAAAAAAAAAGGAACTAAAAAAAG...\n",
       "..                                                ...\n",
       "62  GTGACGACGATATACAGATCCTATATTCTTCATGTATGTAATAGAT...\n",
       "63  GCAGTAAAAATAGGGTAAACACAGGTAAAAGACGGTATGGAAACGG...\n",
       "64  GAAAAAAATGCAGACCTTTCTTAATTCAACTTTTTTTCTTTCAAAT...\n",
       "65  TTGAGTACAGTGGGACGATGTTTCAAAGGTCTGGCGCTGCTCATCA...\n",
       "66  GGTGGTGATAAAGGGAAAATGCGCGCGGGATTAGCAAAGGCGGCCA...\n",
       "\n",
       "[67 rows x 1 columns]"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nonoverlapping_cut_sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "sequence    67\n",
       "dtype: int64"
      ]
     },
     "execution_count": 122,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nonoverlapping_cut_sequences.nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "sequence    843\n",
       "dtype: int64"
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "overlapping_cut_sequences.nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "sequence    219\n",
       "dtype: int64"
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "noncoding_sequences.nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GGGCCCTTTCTTCCGTTTGAACGTAAAGGCATTTTTGAGACCATTA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GTGAATGATGAATTTAATTCTTTGGTCCGTGTTTATGATGGGAAGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GGTTGTTTGGCCGAGCGGTCTAAGGCGCCTGATTCAAGAAATATCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GGGCGTGTGGTCTAGTGGTATGATTCTCGCTTTGGGCGACTTCCTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GGGCACATGGCGCAGTTGGTAGCGCGCTTCCCTTGCAAGGAAGAGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>424</th>\n",
       "      <td>GTGCTACTAAAGTTGTGCGTTTCAGTTAGATTTCTCAAAGGCATTA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>GTCGACGTACTTCAGTATGTTTTATACCATATACTTTATTAGGAAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>426</th>\n",
       "      <td>TATTATGATGATTTTTTTATATTCACACTGTACTAGATTGGTCTCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>427</th>\n",
       "      <td>TGATATGATGATTTGTTGTCGACCGGGCGGACATATTAGTATCTGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>428</th>\n",
       "      <td>TCCCCTATGATTAAAAATTATTAATCATATACCAATTGTTCCGACT...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>219 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              sequence\n",
       "0    GGGCCCTTTCTTCCGTTTGAACGTAAAGGCATTTTTGAGACCATTA...\n",
       "1    GTGAATGATGAATTTAATTCTTTGGTCCGTGTTTATGATGGGAAGT...\n",
       "2    GGTTGTTTGGCCGAGCGGTCTAAGGCGCCTGATTCAAGAAATATCT...\n",
       "3    GGGCGTGTGGTCTAGTGGTATGATTCTCGCTTTGGGCGACTTCCTG...\n",
       "4    GGGCACATGGCGCAGTTGGTAGCGCGCTTCCCTTGCAAGGAAGAGG...\n",
       "..                                                 ...\n",
       "424  GTGCTACTAAAGTTGTGCGTTTCAGTTAGATTTCTCAAAGGCATTA...\n",
       "425  GTCGACGTACTTCAGTATGTTTTATACCATATACTTTATTAGGAAT...\n",
       "426  TATTATGATGATTTTTTTATATTCACACTGTACTAGATTGGTCTCT...\n",
       "427  TGATATGATGATTTGTTGTCGACCGGGCGGACATATTAGTATCTGT...\n",
       "428  TCCCCTATGATTAAAAATTATTAATCATATACCAATTGTTCCGACT...\n",
       "\n",
       "[219 rows x 1 columns]"
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "noncoding_sequences.drop_duplicates(inplace = True)\n",
    "noncoding_sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 161,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TGTATTTTCCACATAGAAAATTCGATTTTTTTTTTTCAATGCACCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TGACAGATATTCTGCACTTAAAAACTAAAAATATTATACCAACTTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TTCTTTCATGCAATGTGATGTCCATCGGAGAAAACTGTATCTTGTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CGTCAAGATATTTGAAAGTTAATAGACAGTTAACAATAATAACAAC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACGATTGACCAAGTCAGAAAAAAAAAAAAAAAGGAACTAAAAAAAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>905</th>\n",
       "      <td>AAACTTGCCGTTAACAATAAGTGACCTTGCATGGAAGACTGTCGAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>906</th>\n",
       "      <td>TGCTAACTTTTATCATCAAATCAGTAAACCCCTTCTTGTTTTTTCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>907</th>\n",
       "      <td>CACAGATGAGGTCAGATTTAACTATTGACCTTCTTGAAACCAAATA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>908</th>\n",
       "      <td>AACAGGTTCTTTCAACTCTGGAAATTCATCCACAATCTTGTCAGCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>909</th>\n",
       "      <td>GTATAGCTGGTAGCAAACGTAGAAGAACAGAATGGACCAGAAAGTA...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>910 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              sequence\n",
       "0    TGTATTTTCCACATAGAAAATTCGATTTTTTTTTTTCAATGCACCA...\n",
       "1    TGACAGATATTCTGCACTTAAAAACTAAAAATATTATACCAACTTT...\n",
       "2    TTCTTTCATGCAATGTGATGTCCATCGGAGAAAACTGTATCTTGTG...\n",
       "3    CGTCAAGATATTTGAAAGTTAATAGACAGTTAACAATAATAACAAC...\n",
       "4    ACGATTGACCAAGTCAGAAAAAAAAAAAAAAAGGAACTAAAAAAAG...\n",
       "..                                                 ...\n",
       "905  AAACTTGCCGTTAACAATAAGTGACCTTGCATGGAAGACTGTCGAT...\n",
       "906  TGCTAACTTTTATCATCAAATCAGTAAACCCCTTCTTGTTTTTTCT...\n",
       "907  CACAGATGAGGTCAGATTTAACTATTGACCTTCTTGAAACCAAATA...\n",
       "908  AACAGGTTCTTTCAACTCTGGAAATTCATCCACAATCTTGTCAGCA...\n",
       "909  GTATAGCTGGTAGCAAACGTAGAAGAACAGAATGGACCAGAAAGTA...\n",
       "\n",
       "[910 rows x 1 columns]"
      ]
     },
     "execution_count": 161,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_cuts_sequences = nonoverlapping_cut_sequences.append(overlapping_cut_sequences)\n",
    "merged_cuts_sequences.reset_index(drop = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0      False\n",
       "1      False\n",
       "2      False\n",
       "3      False\n",
       "4      False\n",
       "       ...  \n",
       "838    False\n",
       "839    False\n",
       "840    False\n",
       "841    False\n",
       "842    False\n",
       "Length: 910, dtype: bool"
      ]
     },
     "execution_count": 178,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_cuts_sequences.duplicated()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_cuts_ncoding = merged_cuts_sequences.drop_duplicates().append(noncoding_sequences)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TGTATTTTCCACATAGAAAATTCGATTTTTTTTTTTCAATGCACCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TGACAGATATTCTGCACTTAAAAACTAAAAATATTATACCAACTTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TTCTTTCATGCAATGTGATGTCCATCGGAGAAAACTGTATCTTGTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CGTCAAGATATTTGAAAGTTAATAGACAGTTAACAATAATAACAAC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACGATTGACCAAGTCAGAAAAAAAAAAAAAAAGGAACTAAAAAAAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1120</th>\n",
       "      <td>GTGCTACTAAAGTTGTGCGTTTCAGTTAGATTTCTCAAAGGCATTA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1121</th>\n",
       "      <td>GTCGACGTACTTCAGTATGTTTTATACCATATACTTTATTAGGAAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1122</th>\n",
       "      <td>TATTATGATGATTTTTTTATATTCACACTGTACTAGATTGGTCTCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1123</th>\n",
       "      <td>TGATATGATGATTTGTTGTCGACCGGGCGGACATATTAGTATCTGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1124</th>\n",
       "      <td>TCCCCTATGATTAAAAATTATTAATCATATACCAATTGTTCCGACT...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1125 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               sequence\n",
       "0     TGTATTTTCCACATAGAAAATTCGATTTTTTTTTTTCAATGCACCA...\n",
       "1     TGACAGATATTCTGCACTTAAAAACTAAAAATATTATACCAACTTT...\n",
       "2     TTCTTTCATGCAATGTGATGTCCATCGGAGAAAACTGTATCTTGTG...\n",
       "3     CGTCAAGATATTTGAAAGTTAATAGACAGTTAACAATAATAACAAC...\n",
       "4     ACGATTGACCAAGTCAGAAAAAAAAAAAAAAAGGAACTAAAAAAAG...\n",
       "...                                                 ...\n",
       "1120  GTGCTACTAAAGTTGTGCGTTTCAGTTAGATTTCTCAAAGGCATTA...\n",
       "1121  GTCGACGTACTTCAGTATGTTTTATACCATATACTTTATTAGGAAT...\n",
       "1122  TATTATGATGATTTTTTTATATTCACACTGTACTAGATTGGTCTCT...\n",
       "1123  TGATATGATGATTTGTTGTCGACCGGGCGGACATATTAGTATCTGT...\n",
       "1124  TCCCCTATGATTAAAAATTATTAATCATATACCAATTGTTCCGACT...\n",
       "\n",
       "[1125 rows x 1 columns]"
      ]
     },
     "execution_count": 164,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_cuts_ncoding.reset_index(drop = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "sequence    1125\n",
       "dtype: int64"
      ]
     },
     "execution_count": 165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_cuts_ncoding.nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TGTATTTTCCACATAGAAAATTCGATTTTTTTTTTTCAATGCACCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CTTGTCGAAAATACGCGGTGTAGGGAGTTATGGTGGATAACTTTTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AATTACTTCCCTGGCTCGCTCCTCCACTGCCTGGGTAAATTGTTCC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ATTTTGGTAAATGCACACGGATAAATGCATGATATATAAAATATTA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AATGGTAGTCATATCATGTCAAGAATAGGTATCCAAAACGCAGCGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1057</th>\n",
       "      <td>GTGCTACTAAAGTTGTGCGTTTCAGTTAGATTTCTCAAAGGCATTA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1058</th>\n",
       "      <td>GTCGACGTACTTCAGTATGTTTTATACCATATACTTTATTAGGAAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059</th>\n",
       "      <td>TATTATGATGATTTTTTTATATTCACACTGTACTAGATTGGTCTCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1060</th>\n",
       "      <td>TGATATGATGATTTGTTGTCGACCGGGCGGACATATTAGTATCTGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1061</th>\n",
       "      <td>TCCCCTATGATTAAAAATTATTAATCATATACCAATTGTTCCGACT...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1062 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                               sequence\n",
       "0     TGTATTTTCCACATAGAAAATTCGATTTTTTTTTTTCAATGCACCA...\n",
       "1     CTTGTCGAAAATACGCGGTGTAGGGAGTTATGGTGGATAACTTTTT...\n",
       "2     AATTACTTCCCTGGCTCGCTCCTCCACTGCCTGGGTAAATTGTTCC...\n",
       "3     ATTTTGGTAAATGCACACGGATAAATGCATGATATATAAAATATTA...\n",
       "4     AATGGTAGTCATATCATGTCAAGAATAGGTATCCAAAACGCAGCGG...\n",
       "...                                                 ...\n",
       "1057  GTGCTACTAAAGTTGTGCGTTTCAGTTAGATTTCTCAAAGGCATTA...\n",
       "1058  GTCGACGTACTTCAGTATGTTTTATACCATATACTTTATTAGGAAT...\n",
       "1059  TATTATGATGATTTTTTTATATTCACACTGTACTAGATTGGTCTCT...\n",
       "1060  TGATATGATGATTTGTTGTCGACCGGGCGGACATATTAGTATCTGT...\n",
       "1061  TCCCCTATGATTAAAAATTATTAATCATATACCAATTGTTCCGACT...\n",
       "\n",
       "[1062 rows x 1 columns]"
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_cutsoverlap_noncoding = overlapping_cut_sequences.append(noncoding_sequences)\n",
    "merged_cutsoverlap_noncoding.reset_index(drop = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "sequence    1062\n",
       "dtype: int64"
      ]
     },
     "execution_count": 167,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_cutsoverlap_noncoding.nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TGTATTTTCCACATAGAAAATTCGATTTTTTTTTTTCAATGCACCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TGACAGATATTCTGCACTTAAAAACTAAAAATATTATACCAACTTT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TTCTTTCATGCAATGTGATGTCCATCGGAGAAAACTGTATCTTGTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CGTCAAGATATTTGAAAGTTAATAGACAGTTAACAATAATAACAAC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACGATTGACCAAGTCAGAAAAAAAAAAAAAAAGGAACTAAAAAAAG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>281</th>\n",
       "      <td>GTGCTACTAAAGTTGTGCGTTTCAGTTAGATTTCTCAAAGGCATTA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>282</th>\n",
       "      <td>GTCGACGTACTTCAGTATGTTTTATACCATATACTTTATTAGGAAT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>283</th>\n",
       "      <td>TATTATGATGATTTTTTTATATTCACACTGTACTAGATTGGTCTCT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284</th>\n",
       "      <td>TGATATGATGATTTGTTGTCGACCGGGCGGACATATTAGTATCTGT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285</th>\n",
       "      <td>TCCCCTATGATTAAAAATTATTAATCATATACCAATTGTTCCGACT...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>286 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              sequence\n",
       "0    TGTATTTTCCACATAGAAAATTCGATTTTTTTTTTTCAATGCACCA...\n",
       "1    TGACAGATATTCTGCACTTAAAAACTAAAAATATTATACCAACTTT...\n",
       "2    TTCTTTCATGCAATGTGATGTCCATCGGAGAAAACTGTATCTTGTG...\n",
       "3    CGTCAAGATATTTGAAAGTTAATAGACAGTTAACAATAATAACAAC...\n",
       "4    ACGATTGACCAAGTCAGAAAAAAAAAAAAAAAGGAACTAAAAAAAG...\n",
       "..                                                 ...\n",
       "281  GTGCTACTAAAGTTGTGCGTTTCAGTTAGATTTCTCAAAGGCATTA...\n",
       "282  GTCGACGTACTTCAGTATGTTTTATACCATATACTTTATTAGGAAT...\n",
       "283  TATTATGATGATTTTTTTATATTCACACTGTACTAGATTGGTCTCT...\n",
       "284  TGATATGATGATTTGTTGTCGACCGGGCGGACATATTAGTATCTGT...\n",
       "285  TCCCCTATGATTAAAAATTATTAATCATATACCAATTGTTCCGACT...\n",
       "\n",
       "[286 rows x 1 columns]"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_cutsnonoverlap_noncoding = nonoverlapping_cut_sequences.append(noncoding_sequences)\n",
    "merged_cutsnonoverlap_noncoding.reset_index(drop = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "sequence    286\n",
       "dtype: int64"
      ]
     },
     "execution_count": 170,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_cutsnonoverlap_noncoding.nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {},
   "outputs": [],
   "source": [
    "nonoverlapping_cut_sequences.to_csv(\"nonoverlapping_cut_sequences.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {},
   "outputs": [],
   "source": [
    "overlapping_cut_sequences.to_csv(\"overlapping_cut_sequences.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "metadata": {},
   "outputs": [],
   "source": [
    "noncoding_sequences.drop_duplicates().reset_index(drop=True).to_csv(\"noncoding_sequences.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 188,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/bash: wget: command not found\n",
      "chmod: Anaconda3-5.1.0-Linux-x86_64.sh: No such file or directory\n",
      "bash: ./Anaconda3-5.1.0-Linux-x86_64.sh: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "!wget -c https://repo.continuum.io/archive/Anaconda3-5.1.0-Linux-x86_64.sh\n",
    "\n",
    "!chmod +x Anaconda3-5.1.0-Linux-x86_64.sh\n",
    "\n",
    "!bash ./Anaconda3-5.1.0-Linux-x86_64.sh -b -f -p /usr/local"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
